walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,78 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
 
+/** Finds options for times to schedule a meeting request */
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // no options for too long of a request
+    if(request.getDuration() > TimeRange.WHOLE_DAY.duration())
+    {
+        Collection<TimeRange> noTimes = Arrays.asList();
+        return noTimes;
+    }
+
+    Collection<String> attendees = new HashSet<>();
+    attendees = request.getAttendees();
+
+    // initialize options as the whole day
+    Collection<TimeRange> times = new ArrayList<>();
+    times.add(TimeRange.WHOLE_DAY);
+
+    // if there are no attendees the meeting can be any time of the day 
+    if(attendees.isEmpty()){
+        return times; 
+    }
+    
+    for(Event event : events){
+
+        Collection<String> eventAttendees = new HashSet<>();
+        eventAttendees = event.getAttendees();
```

Comment:
This can be one line, with no `new HashSet<>()`. The `new HashSet<>()` value just gets thrown away when you assign a different value to `eventAttendeeds`.

But also, this variable is only used once, so let's just write `event.getAttendees()` inline instead

Reply:
Sounds good I just used event.getAttendees()